brave-instrumentation/aws-java-sdk-sqs-v2/src/main/java/brave/instrumentation/aws/sqs/SendMessageBatchTracingExecutionInterceptor.java

```diff
@@ -0,0 +1,274 @@
+    if (individualMessageSpans == null) {
+      // someone deleted our attribute...
+      return;
+    }
+
+    if (!context.httpResponse().isSuccessful()) {
+      individualMessageSpans.values()
+          .forEach(
+              span -> {
+                try {
+                  spanDecorator.decorateRequestFailedMessageSpan(request,
+                      context.httpResponse(), span);
+                  span.error(new RuntimeException("Error placing message onto SQS queue"));
+                } finally {
+                  span.finish();
+                }
+              });
+      return;
+    }
+
+    final SendMessageBatchResponse response = (SendMessageBatchResponse) context.response();
+
+    response.successful().forEach(result -> {
+      final Span messageSpan = individualMessageSpans.get(result.id());
+      if (messageSpan == null) {
+        // for some reason the individual message's span cannot be found
+        return;
+      }
+
+      try {
+        spanDecorator.decorateMessageSuccessfulSpan(response, context.httpResponse(),
+            result, messageSpan);
+      } finally {
+        messageSpan.finish();
+      }
+    });
+
+    response.failed().forEach(result -> {
+      final Span messageSpan = individualMessageSpans.get(result.id());
+      if (messageSpan == null) {
+        // for some reason the individual message's span cannot be found
+        return;
+      }
+
+      try {
+        spanDecorator.decorateMessageFailureSpan(response, context.httpResponse(), result,
+            messageSpan);
+        messageSpan.error(new RuntimeException("Error placing message onto SQS queue"));
+      } finally {
+        messageSpan.finish();
+      }
+    });
+  }
+
+  private Span startSpanForMessage(final SendMessageBatchRequest request,
+      final SendMessageBatchRequestEntry requestEntry) {
+    final Span messageSpan = tracing.tracer().nextSpan();
+    if (!messageSpan.isNoop()) {
+      spanDecorator.decorateMessageSpan(request, requestEntry, messageSpan);
+    }
+
+    messageSpan.start();
+    return messageSpan;
+  }
+
+  private SendMessageBatchRequestEntry injectSpanInformationIntoMessage(
+      final SendMessageBatchRequestEntry entry, final Span span) {
+    final Map<String, MessageAttributeValue> currentMessageAttributes =
+        new HashMap<>(entry.messageAttributes());
+
+    messageAttributeInjector.inject(span.context(), currentMessageAttributes);
+
+    return entry.toBuilder()
+        .messageAttributes(currentMessageAttributes)
+        .build();
+  }
+
+  /**
+   * Decorator that can be used to override the default span configurations, otherwise the {@link
+   * SpanDecorator#DEFAULT} can be used.
+   *
+   * <p>This would be helpful if you want to send extra information or remove some of the tags
+   * being sent as they are not relevant for your use case.
+   */
+  @SuppressWarnings("unused")
+  public interface SpanDecorator {
```

Comment:
This seemed like the easiest way to provide defaults but also allowing consumers to override an individual method with their functionality.